test/prism/fixtures/it_assignment.txt

```diff
@@ -0,0 +1 @@
+42.tap { it = it; p it }
```

Comment:
Maybe also test `42.tap { it; it = it; p it }`? This should still generate a it read for the first one.

Reply:
Thank you for the comment.

You're right, that test case would be good to have as well.
I've added the test case in the following commit: https://github.com/ruby/prism/pull/3604/commits/659d769621a461a889772aa37a5d9dbf0f8743fe